packages/actions/src/internal/runCallWithPrestateTrace.js

```diff
@@ -0,0 +1,203 @@
+import { createAddress } from '@tevm/address'
+import { DefensiveNullCheckError } from '@tevm/errors'
+import { bytesToHex, toHex } from '@tevm/utils'
+
+/**
+ * @internal
+ * Captures the state of an account, including its storage based on accessed slots
+ * @param {import('@tevm/vm').Vm} vm
+ * @param {import('@tevm/node').TevmNode['logger']} logger
+ * @param {import('@tevm/utils').EthjsAddress} address
+ * @param {Set<import('@tevm/utils').Hex>} slots Set of storage slot keys that were accessed
+ * @returns {Promise<import('../debug/DebugResult.js').AccountState | null>}
+ */
+const captureAccountState = async (vm, logger, address, slots = new Set()) => {
+	try {
+		// Get account details
+		const account = await vm.stateManager.getAccount(address)
+		if (!account) return null
+
+		// Get code
+		const code = await vm.stateManager.getContractCode(address)
+
+		// Get storage for all accessed slots
+		/** @type {Record<import('@tevm/utils').Hex, import('@tevm/utils').Hex>} */
+		const storage = {}
+		for (const slotHex of slots) {
+			try {
+				// Convert hex slot to Uint8Array for getContractStorage
+				const slotKey = Buffer.from(slotHex.slice(2), 'hex')
+				const value = await vm.stateManager.getContractStorage(address, slotKey)
+				storage[slotHex] = bytesToHex(value)
+			} catch (err) {
+				logger.error(err, `Error getting storage at slot ${slotHex} for account ${address.toString()}`)
+			}
+		}
+
+		return {
+			balance: toHex(account.balance),
+			nonce: account.nonce.toString(),
+			code: code && code.length > 0 ? bytesToHex(code) : '0x',
+			storage,
+		}
+	} catch (err) {
+		logger.error(err, `Error capturing state for account ${address.toString()}`)
+		return null
+	}
+}
+
+/**
+ * @internal
+ * Executes a call with prestate tracer, which captures account state before and after execution
+ * @template {boolean} TDiffMode
+ * @param {import('@tevm/vm').Vm} vm
+ * @param {import('@tevm/node').TevmNode['logger']} logger
+ * @param {import('@tevm/evm').EvmRunCallOpts} params
+ * @param {TDiffMode} diffMode If true, only returns state that changed between pre and post execution
+ * @returns {Promise<import('@tevm/evm').EvmResult & {trace: TDiffMode extends true ? import('../debug/DebugResult.js').PrestateTracerDiffResult : import('../debug/DebugResult.js').PrestateTracerResult}>}
+ * @throws {never}
+ */
+export const runCallWithPrestateTrace = async (vm, logger, params, diffMode = /** @type {TDiffMode} */ (false)) => {
+	// Prepare VM for tracing
+	await vm.evm.journal.cleanup()
+	// Start tracking which accounts and storage slots are accessed during execution
+	vm.evm.journal.startReportingAccessList()
```

Comment:
```suggestion
	vm.evm.journal.startReportingAccessList()
  vm.evm.journal.startReportingPreimages
```
You can get preimages this way too